### ISSUE TITLE
Release Candidate 2 BUG REPORTS

### ISSUE THREAD

**Issue by liversedge** [GoldenCheetah/GoldenCheetah#1424]
Bugs highlighted as part of 3.2 RC2 release testing

**Comment by liversedge** [comment]
Skiba PMC differs between 3.1 and 3.2

**Comment by liversedge** [comment]
Moxy Firmware 1.0 has come out and should be explicitly supported since it means the data is in a different format.

**Comment by liversedge** [comment]
Adding / removing columns seems to do something weird with groupBy in the ride list.

**Comment by amtriathlon** [comment]
Removing speed column in Editor doesn't trigger metric re-computation, Clear Contents does.

**Comment by liversedge** [comment]
LTM data table metric to imperial conversion is missed e.g. Trends > Tracker

**Comment by liversedge** [comment]
compare by distance for short durations x-axis is condensed in allplot

**Comment by amtriathlon** [comment]
Metric to imperial conversion is done twice in Trends->RideSummary, visible in Distance as reported in the forum.

**Comment by Joern-R** [comment]
Introduction of "Multi-Sport" Ride Files through Commit: https://github.com/GoldenCheetah/GoldenCheetah/commit/7fdedf0bfeb977f94cd789826bee9352458d3208
assumes an "interleaved" sequence of session, record and even messages. Problem is that FIT allows both "interleaved" structure (which is assumed in the code) and grouping session and other summary message grouped together at the beginning of the file. (While e.g. Garmin seems to use the Interleaved way (and the code works). E.g. O_Synce put's the session message (for the Ride) at the beginning (before any events). 

This leads to an mostly empty ride file (with no date, length,...) when test-importing/checking the file. While the final import then works fine. Since the behaviour if file checking in imports deviates from the final import I am also not sure if multi-sessions are properly processed.

I have an O_SYNCE file available for testing - just let me know via mail.

(Not sure how to best fix this / and I have not test files for multi-session available to check any side effect of changes. - Sorry)

**Comment by liversedge** [comment]
I may revert the multi-sport commit until we can decide how best to handle this.

**Comment by liversedge** [comment]
SEGV on compare season containing athlete weight

**Comment by liversedge** [comment]
SEGV when dragging newly created season into compare pane.

**Comment by amtriathlon** [comment]
In CP chart, when changing model to Ward-Smith, search intervals aren't updated and stay as they were for the previously selected model.

**Comment by liversedge** [comment]
**SEGV** in LTMPlot.

```
Crash at LTMPlot.cpp:1125           current->setSamples(xdata.data(),ydata.data(), count + 1);

Back trace (I added some debug so line numbers are offset by 6)

#0  __memcpy_sse2_unaligned () at ../sysdeps/x86_64/multiarch/memcpy-sse2-unaligned.S:36
#1  0x0000000000a63161 in QwtPointArrayData::QwtPointArrayData(double const*, double const*, unsigned long) ()
#2  0x0000000000a492f2 in QwtPlotCurve::setSamples(double const*, double const*, int) ()
#3  0x00000000006e96af in LTMPlot::setData (this=0x45f7550, set=set@entry=0x45ef3b0) at LTMPlot.cpp:1131
#4  0x0000000000728c37 in LTMWindow::refreshPlot (this=this@entry=0x45ef1c0) at LTMWindow.cpp:466
#5  0x000000000072c4c7 in LTMWindow::groupBySelected (this=this@entry=0x45ef1c0, selected=1) at LTMWindow.cpp:855
#6  0x00000000009d741f in LTMWindow::qt_static_metacall (_o=0x45ef1c0, _c=<optimised out>, _id=<optimised out>, _a=0x7fffffffce70) at moc_LTMWindow.cpp:130
#7  0x00007ffff39e087a in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#8  0x00007ffff460f7c1 in QComboBox::currentIndexChanged(int) () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#9  0x00007ffff460f821 in ?? () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#10 0x00007ffff460facc in ?? () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#11 0x00007ffff460fd1f in QComboBox::setCurrentIndex(int) () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#12 0x0000000000723b43 in LTMWindow::rGroupBySelected (this=this@entry=0x45ef1c0, selected=1) at LTMWindow.cpp:845
#13 0x00000000009d744f in LTMWindow::qt_static_metacall (_o=0x45ef1c0, _c=<optimised out>, _id=<optimised out>, _a=0x7fffffffd140) at moc_LTMWindow.cpp:132
#14 0x00007ffff39e087a in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#15 0x00007ffff489837e in QSpinBox::valueChanged(int) () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#16 0x00007ffff4686d86 in ?? () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#17 0x00007ffff45fe11e in ?? () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#18 0x00007ffff45fef42 in QAbstractSpinBox::stepBy(int) () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#19 0x00007ffff45fd258 in ?? () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#20 0x00007ffff45fd3cf in QAbstractSpinBox::mousePressEvent(QMouseEvent*) () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#21 0x00007ffff429538b in QWidget::event(QEvent*) () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#22 0x00007ffff4245e2c in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#23 0x00007ffff424c5dd in QApplication::notify(QObject*, QEvent*) () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#24 0x00007ffff39cc4dd in QCoreApplication::notifyInternal(QObject*, QEvent*) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#25 0x00007ffff424bd93 in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool) () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#26 0x00007ffff42c09eb in ?? () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#27 0x00007ffff42c0289 in QApplication::x11ProcessEvent(_XEvent*) () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#28 0x00007ffff42e7b32 in ?? () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#29 0x00007ffff0812e04 in g_main_context_dispatch () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#30 0x00007ffff0813048 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#31 0x00007ffff08130ec in g_main_context_iteration () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#32 0x00007ffff39f97a1 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#33 0x00007ffff42e7be6 in ?? () from /usr/lib/x86_64-linux-gnu/libQtGui.so.4
#34 0x00007ffff39cb0af in QEventLoop::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#35 0x00007ffff39cb3a5 in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#36 0x00007ffff39d0b79 in QCoreApplication::exec() () from /usr/lib/x86_64-linux-gnu/libQtCore.so.4
#37 0x000000000045236c in main (argc=1, argv=<optimised out>) at main.cpp:338


Reproducer:
- Select trends->PMC
- Select 'This week' (no actual rides this week yet)
- Select Group by 'weeks'
- Seg fault

I added debug into LTMPlot.cpp:

sbrewer@kingfisher:~/Goldencheetah/GIT$ git diff src/LTMPlot.cpp
diff --git a/src/LTMPlot.cpp b/src/LTMPlot.cpp
index 17f9c6b..f30ae0b 100644
--- a/src/LTMPlot.cpp
+++ b/src/LTMPlot.cpp
@@ -1122,6 +1122,12 @@ LTMPlot::setData(LTMSettings *set)
         }
 
         // set the data series
+qDebug() << "xdata" << xdata.data();
+qDebug() << "ydata" << ydata.data();
+qDebug() << "count" << count;
+qDebug() << "xd.count" << xdata.count();
+qDebug() << "yd.count" << ydata.count();
+
         current->setSamples(xdata.data(),ydata.data(), count + 1);
         current->setBaseline(metricDetail.baseline);
 

And got this:

xdata 0x5775c90 
ydata 0x5780730 
count 6 
xd.count 9 
yd.count 9 
xdata 0x58f6950 
ydata 0x5a70450 
count 6 
xd.count 9 
yd.count 9 
xdata 0x58ea080 
ydata 0x4b722a0 
count 6 
xd.count 9 
yd.count 9 
xdata 0x5d13be0 
ydata 0x5d13dc0 
count 32 
xd.count 36 
yd.count 36 
xdata 0x48fdd40 
ydata 0x48fdc20 
count 20761696 
xd.count 0 
yd.count 0 


'Count' seems screwed up.

thanks,

Simon
```

**Comment by gcoco** [comment]
When sorting rides by Distance the sorting is done by string and not numerically.

![image](https://cloud.githubusercontent.com/assets/925347/8631629/a64ab444-2749-11e5-86a1-f745c0f4d9aa.png)

**Comment by amtriathlon** [comment]
When there is no Temp series AvgTemp and MaxTemp show -255 Celsius or -427 Farenheit

**Comment by liversedge** [comment]
~~Classic and Morton PD models /kg estimates not being saved.~~ Fixed in 4694109149c7b18dabee6285bfe20e48627c222b
